src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -526,6 +584,7 @@ public void startedGesture(String nameOfGesture) {
     if (gestureAlreadyStarted) {
       warn("Warning 1 gesture already running, this can break spacetime and lot of things");
     } else {
+      log.info("Starting gesture : {}", nameOfGesture);
       gestureAlreadyStarted = true;
```

Comment:
seems like a lot of booleans here to maintain state.  This logic is too complex and should be simplified.  Having so many state variables is very difficult to test and leads to bugs.